### PR TITLE
[language] add support for declaring validator accounts in functional…

### PR DESCRIPTION
**Pull request by sblackshear**
… tests

Testing features like reconfiguration and validator payments requires knowing which accounts belong to validators. This PR allows functional tests to declare validator accounts (e.g., "account: alice, 1000, 0, validator"). The testing framework will generate a validator set with either the number of declared validators (if any are declared) or, in the absence of declared validators, use the default behavior (currently, a validator set of size 10).

In testing this feature I found/fixed a bug in ValidatorSet.is_validator.

## Motivation

See above--needed to test features that involve on-chain representation of validators.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Indeed.

## Test Plan

New E2E tests.

## Related PRs

None.

### CODE REVIEW

File: language/functional_tests/src/config/global.rs [hunk 1]
```diff
@@ -70,18 +105,49 @@ pub struct Config {
...
 
 impl Config {
     pub fn build(entries: &[Entry]) -> Result<Self> {
         let mut accounts = BTreeMap::new();
+        let mut validator_accounts = 0;
+        for entry in entries {
+            if entry.is_validator() {
+                validator_accounts += 1;
+            }
+        }
```
Comment (vgao1996) [inline language/functional_tests/src/config/global.rs]: `let mut validator_accounts = entries.filter(|entry| entry.is_validator()).count();`?

File: language/functional_tests/src/config/global.rs [hunk 2]
```diff
@@ -70,18 +105,49 @@ pub struct Config {
...
             match entry {
                 Entry::AccountDefinition(def) => {
-                    let account_data = AccountData::new(
-                        def.balance.unwrap_or(DEFAULT_BALANCE),
-                        def.sequence_number.unwrap_or(0),
-                    );
+                    let account_data = if entry.is_validator() {
+                        validator_accounts -= 1;
+                        let validator_public_keys =
+                            validator_set.payload()[validator_accounts].clone();
```
Comment (vgao1996) [inline language/functional_tests/src/config/global.rs]: The counter `validator_accounts` seems funny to me. Just curious why this needs to be in reverse order?

File: language/functional_tests/src/tests/global_config_tests.rs [hunk 3]
```diff
@@ -27,7 +27,11 @@ fn parse_account_positive() {
 
 #[test]
 fn parse_account_negative() {
```
Comment (vgao1996) [inline language/functional_tests/src/tests/global_config_tests.rs]: Also add a positive test case? Just to make sure "//! account: alice, 1, 2, validator" parses.

File: language/functional_tests/src/config/global.rs (hunk 1 above)
Comment (sblackshear) [reply language/functional_tests/src/config/global.rs]: Thanks! I knew there should be a better Rust-y way to do this..

File: language/functional_tests/src/config/global.rs (hunk 2 above)
Comment (sblackshear) [reply language/functional_tests/src/config/global.rs]: There's no need for it to be in reverse order; we could create a second counter that starts from 0 and goes up instead. I did it this way to reuse the first counter + make the relationship between the validator set indexes and the initial counter clear.

**Pull request closed without merging**